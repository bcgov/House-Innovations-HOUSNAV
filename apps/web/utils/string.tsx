// 3rd party
import { FunctionComponent, ReactNode } from "react";
import parse, {
  DOMNode,
  Element,
  domToReact,
  attributesToProps,
} from "html-react-parser";
// local
import DefinedTerm, {
  DefinedTermProps,
} from "../components/defined-term/DefinedTerm";
import Button from "@repo/ui/button";

// Define custom components for html-react-parser
const definedTermName = "defined-term";
const definedTermModal = "defined-term-modal";
type CustomComponentTypes = typeof definedTermName | typeof definedTermModal;
type CustomComponentProps = DefinedTermProps;
const customComponents: Record<
  CustomComponentTypes,
  FunctionComponent<CustomComponentProps>
> = {
  [definedTermName]: DefinedTerm,
  [definedTermModal]: Button,
};

export const parseStringToComponents = (html: string, customHandler?: any) => {
  const options = {
    replace: (domNode: DOMNode) => {
      if (domNode instanceof Element && domNode.attribs) {
        switch (domNode.name) {
          case definedTermName: {
            const DefinedTermComponent = customComponents[definedTermName];
            const props = attributesToProps(domNode.attribs);

            if (DefinedTermComponent) {
              return (
                <DefinedTermComponent
                  {...(props as unknown as DefinedTermProps)}
                  key={domNode.attribs.key}
                >
                  {domToReact(domNode.children as DOMNode[], options)}
                </DefinedTermComponent>
              );
            }
          }
          case definedTermModal: {
            // TODO: Matt or Nicholas, add correct section for customHandler
            const randomSection = Math.floor(Math.random() * 10) + 1;
            return (
              <Button
                variant="glossary"
                onPress={
                  customHandler
                    ? () => customHandler(`9.9.9.${randomSection}`)
                    : () => {}
                }
              >
                {domToReact(domNode.children as DOMNode[])}
              </Button>
            );
          }
        }
      }
    },
  };
  return parse(html, options);
};

export const getStringFromComponents = (node: ReactNode): string => {
  if (node == null) return "";

  switch (typeof node) {
    case "string":
    case "number":
      return node.toString();

    case "boolean":
      return "";

    case "object": {
      if (Array.isArray(node))
        return node.map(getStringFromComponents).join("");

      if ("props" in node) return getStringFromComponents(node.props.children);
    } // eslint-ignore-line no-fallthrough

    default:
      // NOTE: I don't believe this is reachable, but it's here for completeness and to avoid a linter warning
      console.warn("Unresolved `node` of type:", typeof node, node);
      return "";
  }
};
