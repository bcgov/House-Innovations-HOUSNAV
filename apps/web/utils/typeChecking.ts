// NOTE: this feels weird, but it makes sure this module is something we can spy on for tests
import * as ThisModule from "./typeChecking";

export const isString = (value: unknown): value is string => {
  return typeof value === "string";
};

export const isArray = (value: unknown): value is unknown[] => {
  return Array.isArray(value);
};

export const isObject = (value: unknown): value is Record<string, unknown> => {
  return typeof value === "object" && !ThisModule.isArray(value);
};

export const isNumber = (value: unknown): value is number => {
  return typeof value === "number";
};
