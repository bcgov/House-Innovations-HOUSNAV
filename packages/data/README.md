# `@repo/data`

JSON data files for the `@repo` package plus interfaces to access them. This contains all current data for each walkthrough, the defined terms, and the digital building code content.

## Walkthrough Schema

All walkthrough type JSON files follow the same schema outlined in `schema/walkthrough.schema.json`. Each JSON with the `wt-` prefix follows this schema. They point to the schema file for validation and their associated unit tests verify data not verified by the schema.
