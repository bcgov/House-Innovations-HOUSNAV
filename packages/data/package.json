{
  "name": "@repo/data",
  "version": "0.0.0",
  "private": true,
  "exports": {
    "./useData": "./src/useData.ts",
    "./useWalkthroughData": "./src/useWalkthroughData.ts",
    "./useWalkthroughTestData": "./src/useWalkthroughTestData.ts"
  },
  "scripts": {
    "lint": "eslint . --max-warnings 0",
    "format": "prettier --write \"**/*.{ts,tsx,md,css}\"",
    "format:check": "prettier --check \"**/*.{ts,tsx,md,css}\""
  },
  "devDependencies": {
    "@repo/eslint-config": "*",
    "@repo/typescript-config": "*",
    "@types/eslint": "^8.56.5",
    "eslint": "^8.57.0",
    "typescript": "^5.3.3"
  }
}
