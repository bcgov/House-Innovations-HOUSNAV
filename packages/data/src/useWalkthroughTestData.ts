import testCase999 from "../walkthroughes/9.9.9.json";
import {
  QuestionData,
  QuestionMultipleChoiceData,
  WalkthroughItemTypeMultiChoice,
} from "./useWalkthroughData";

export function getMultiChoiceQuestion() {
  // find the first question that is a multi-choice question
  const questions: { [key: string]: QuestionData } = testCase999.questions;
  const questionKey = Object.keys(questions).find((questionId) => {
    const question = questions[questionId] as QuestionData;
    return question.walkthroughItemType === WalkthroughItemTypeMultiChoice;
  });
  return questionKey
    ? {
        questionKey,
        questionData: questions[questionKey] as QuestionMultipleChoiceData,
      }
    : null;
}

// export function getMultiChoiceMultipleQuestion() {
//   // find the first question that is a multi-choice multiple question
//   const questions: { [key: string]: QuestionData } = testCase999.questions;
//   const questionKey = Object.keys(questions).find((questionId) => {
//     const question = questions[questionId] as QuestionData;
//     return (
//       question.walkthroughItemType === WalkthroughItemTypeMultiChoiceMultiple
//     );
//   });
//   return questionKey
//     ? {
//         questionKey,
//         questionData: questions[
//           questionKey
//         ] as QuestionMultipleChoiceSelectMultipleData,
//       }
//     : null;
// }

export default function useWalkthroughTestData() {
  return testCase999;
}
