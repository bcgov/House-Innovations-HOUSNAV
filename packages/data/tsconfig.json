{
  "extends": "@repo/typescript-config/react-library.json",
  "include": ["src", "walkthroughs", "vite.config.mts"],
}
