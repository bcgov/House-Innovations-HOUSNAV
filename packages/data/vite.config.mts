import { defineProject } from "vitest/config";

export default defineProject({
  test: {
    globals: true,
    coverage: {
      include: ["src/**/*"],
    },
  },
});
