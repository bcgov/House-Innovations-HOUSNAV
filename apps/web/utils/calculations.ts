// repo
import {
  PropertyCalculationValueToUse,
  PropertyDefaultValue,
  PropertyNameAnswerToUse,
  PropertyResultLogicItems,
  ResultCalculation,
  ResultCalculationType,
  ResultCalculationValue,
  ResultLogicItem,
  ResultLogicTypes,
  VariableValueCalculationType,
  VariableValueToSetCalculation,
} from "@repo/data/useWalkthroughData";
// local
import { AnswerToCheckValueFn } from "../stores/AnswerStore";
import { isNumber } from "./typeChecking";
// NOTE: this feels weird, but it makes sure this module is something we can spy on for tests
import * as ThisModule from "./calculations";

export const calculateVariableValueToSet = (
  answerValueToUse: number,
  variableValueToSetCalculation: VariableValueToSetCalculation,
): number => {
  switch (variableValueToSetCalculation.variableValueCalculationType) {
    case VariableValueCalculationType.Then:
      if (variableValueToSetCalculation.variableValueToSetItemCalculations) {
        // reduce with the answerValueToUse
        return variableValueToSetCalculation.variableValueToSetItemCalculations.reduce<number>(
          (acc, item) => {
            return ThisModule.calculateVariableValueToSet(
              acc,
              item as VariableValueToSetCalculation,
            );
          },
          answerValueToUse,
        );
      }

      throw new Error(
        "variableValueToSetItemCalculations must exist when variableValueCalculationType is Then",
      );
    case VariableValueCalculationType.Square:
      return ThisModule.calculationTypeSquare(answerValueToUse);
    case VariableValueCalculationType.Multiply:
      if (variableValueToSetCalculation.calculationValueToUse) {
        return ThisModule.calculationTypeMultiply(
          answerValueToUse,
          variableValueToSetCalculation.calculationValueToUse,
        );
      }

      throw new Error(
        "calculationValueToUse must exist when variableValueCalculationType is Multiply",
      );
    case VariableValueCalculationType.Divide:
      if (variableValueToSetCalculation.calculationValueToUse) {
        return ThisModule.calculationTypeDivide(
          answerValueToUse,
          variableValueToSetCalculation.calculationValueToUse,
        );
      }

      throw new Error(
        "calculationValueToUse must exist when variableValueCalculationType is Divide",
      );
    case VariableValueCalculationType.MaxValue:
      if (variableValueToSetCalculation.calculationValueToUse) {
        return Math.min(
          answerValueToUse,
          variableValueToSetCalculation.calculationValueToUse,
        );
      }

      throw new Error(
        "calculationValueToUse must exist when variableValueCalculationType is MaxValue",
      );
  }

  throw new Error(
    `variableValueCalculationType ${variableValueToSetCalculation.variableValueCalculationType} is not supported.`,
  );
};

export const calculateResultDisplayNumber = (
  resultCalculation: ResultCalculation | ResultCalculationValue,
  getAnswerToCheckValue: AnswerToCheckValueFn,
): number => {
  const { resultCalculationType } = resultCalculation;

  switch (resultCalculation.resultCalculationType) {
    case ResultCalculationType.MaxBetween: {
      const { calculationValuesToUse } = resultCalculation;
      if (calculationValuesToUse) {
        const maxBetweenValues = calculationValuesToUse.map(
          (calculationValue) => {
            return ThisModule.calculateResultDisplayNumber(
              calculationValue,
              getAnswerToCheckValue,
            );
          },
        );

        return Math.max(...maxBetweenValues);
      }

      throw new Error(
        `calculateResultValue: type ${ResultCalculationType.MaxBetween} must have calculationValuesToUse.`,
      );
    }
    case ResultCalculationType.MinBetween: {
      const { calculationValuesToUse } = resultCalculation;
      if (calculationValuesToUse) {
        const minBetweenValues = calculationValuesToUse.map(
          (calculationValue) => {
            return ThisModule.calculateResultDisplayNumber(
              calculationValue,
              getAnswerToCheckValue,
            );
          },
        );

        return Math.min(...minBetweenValues);
      }

      throw new Error(
        `calculateResultValue: type ${ResultCalculationType.MinBetween} must have calculationValuesToUse.`,
      );
    }
    case ResultCalculationType.Divide: {
      const { calculationValuesToUse } = resultCalculation;
      if (calculationValuesToUse) {
        const divideValues = calculationValuesToUse.map((calculationValue) => {
          return ThisModule.calculateResultDisplayNumber(
            calculationValue,
            getAnswerToCheckValue,
          );
        });

        if (
          divideValues.length === 2 &&
          isNumber(divideValues[0]) &&
          isNumber(divideValues[1])
        )
          return ThisModule.calculationTypeDivide(
            divideValues[0],
            divideValues[1],
          );

        throw new Error(
          `calculateResultValue: type ${ResultCalculationType.Divide} only supports 2 values and they must be numbers.`,
        );
      }

      throw new Error(
        `calculateResultValue: type ${ResultCalculationType.Divide} must have calculationValuesToUse.`,
      );
    }
    case ResultCalculationType.Multiply: {
      const { calculationValuesToUse } = resultCalculation;
      if (calculationValuesToUse) {
        const multiplyValues = calculationValuesToUse.map(
          (calculationValue) => {
            return ThisModule.calculateResultDisplayNumber(
              calculationValue,
              getAnswerToCheckValue,
            );
          },
        );

        if (
          multiplyValues.length === 2 &&
          isNumber(multiplyValues[0]) &&
          isNumber(multiplyValues[1])
        )
          return ThisModule.calculationTypeMultiply(
            multiplyValues[0],
            multiplyValues[1],
          );

        throw new Error(
          `calculateResultValue: type ${ResultCalculationType.Multiply} only supports 2 values and they must be numbers.`,
        );
      }

      throw new Error(
        `calculateResultValue: type ${ResultCalculationType.Multiply} must have calculationValuesToUse.`,
      );
    }
    case ResultCalculationType.Minus: {
      const { calculationValuesToUse } = resultCalculation;
      if (calculationValuesToUse) {
        const minusValues = calculationValuesToUse.map((calculationValue) => {
          return ThisModule.calculateResultDisplayNumber(
            calculationValue,
            getAnswerToCheckValue,
          );
        });

        if (
          minusValues.length === 2 &&
          isNumber(minusValues[0]) &&
          isNumber(minusValues[1])
        )
          return ThisModule.calculationTypeMinus(
            minusValues[0],
            minusValues[1],
          );

        throw new Error(
          `calculateResultValue: type ${ResultCalculationType.Minus} only supports 2 values and they must be numbers.`,
        );
      }

      throw new Error(
        `calculateResultValue: type ${ResultCalculationType.Minus} must have calculationValuesToUse.`,
      );
    }
    case ResultCalculationType.Square: {
      if (!(PropertyCalculationValueToUse in resultCalculation)) {
        throw new Error(
          `calculateResultValue: type ${ResultCalculationType.Square} must have ${PropertyCalculationValueToUse} property.`,
        );
      }

      const calculationValueToUse =
        resultCalculation[PropertyCalculationValueToUse];
      if (calculationValueToUse && !isNumber(calculationValueToUse)) {
        return ThisModule.calculationTypeSquare(
          ThisModule.calculateResultDisplayNumber(
            calculationValueToUse,
            getAnswerToCheckValue,
          ),
        );
      }

      throw new Error(
        `calculateResultValue: type ${ResultCalculationType.Square} must have ${PropertyCalculationValueToUse} that is not a number.`,
      );
    }
    case ResultCalculationType.Answer: {
      if (
        !(PropertyNameAnswerToUse in resultCalculation) ||
        !resultCalculation[PropertyNameAnswerToUse]
      ) {
        throw new Error(
          `calculateResultValue: type ${ResultCalculationType.Answer} must have ${PropertyNameAnswerToUse} property.`,
        );
      }

      const answerToUseValue = getAnswerToCheckValue(
        resultCalculation[PropertyNameAnswerToUse],
      );
      if (answerToUseValue && isNumber(answerToUseValue))
        return answerToUseValue;

      if (!answerToUseValue && PropertyDefaultValue in resultCalculation) {
        const { defaultValue } = resultCalculation;
        if (isNumber(defaultValue)) return defaultValue;
      }

      throw new Error(
        `calculateResultValue: type ${ResultCalculationType.Answer} answerToUseValue was not a number, or it wasn't found and no ${PropertyDefaultValue} was provided.`,
      );
    }
    case ResultCalculationType.Number: {
      if (!(PropertyCalculationValueToUse in resultCalculation)) {
        throw new Error(
          `calculateResultValue: type ${ResultCalculationType.Number} must have ${PropertyCalculationValueToUse} property.`,
        );
      }

      const calculationValueToUse =
        resultCalculation[PropertyCalculationValueToUse];
      if (isNumber(calculationValueToUse)) return calculationValueToUse;

      throw new Error(
        `calculateResultValue: type ${ResultCalculationType.Number} must have ${PropertyCalculationValueToUse} that is a number.`,
      );
    }
    case ResultCalculationType.Logic: {
      if (
        !(PropertyResultLogicItems in resultCalculation) ||
        !resultCalculation[PropertyResultLogicItems]
      ) {
        throw new Error(
          `calculateResultValue: type ${ResultCalculationType.Logic} must have ${PropertyResultLogicItems} property.`,
        );
      }

      const resultLogicItems = resultCalculation[PropertyResultLogicItems];
      if (resultLogicItems) {
        for (const resultLogicItem of resultLogicItems) {
          if (
            ThisModule.resultLogicTypeIsTrue(
              resultLogicItem,
              getAnswerToCheckValue,
            )
          ) {
            return resultLogicItem.valueToUse;
          }
        }

        throw new Error(
          `calculateResultValue: type ${ResultCalculationType.Logic} found no true resultLogicItem.`,
        );
      }

      throw new Error(
        `calculateResultValue: type ${ResultCalculationType.Logic} must have calculationValuesToUse.`,
      );
    }
  }

  throw new Error(
    `resultCalculationType ${resultCalculationType} is not supported.`,
  );
};

export const resultLogicTypeIsTrue = (
  resultLogicItem: ResultLogicItem,
  getAnswerToCheckValue: AnswerToCheckValueFn,
): boolean => {
  const { resultLogicType } = resultLogicItem;

  // if fallback item is found, return true
  // paradigms dictate that the fallback item should be the last item in the array
  if (resultLogicType === ResultLogicTypes.Fallback) {
    return true;
  }

  if (resultLogicType === ResultLogicTypes.GreaterThan) {
    const { answerValue, answerToCheck } = resultLogicItem;
    if (answerValue && answerToCheck) {
      const answerToCheckValue = getAnswerToCheckValue(answerToCheck);
      if (isNumber(answerToCheckValue)) {
        return answerToCheckValue > answerValue;
      }

      throw new Error(
        `resultLogicType ${ResultLogicTypes.GreaterThan} found no answerToCheckValue, or answerToCheckValue was not a number.`,
      );
    }

    throw new Error(
      `resultLogicType ${ResultLogicTypes.GreaterThan} must have answerValue and answerToCheck.`,
    );
  }

  throw new Error(`resultLogicType ${resultLogicType} is not supported.`);
};

export const calculationTypeSquare = (answerValueToUse: number) => {
  return answerValueToUse * answerValueToUse;
};

export const calculationTypeDivide = (
  numerator: number,
  denominator: number,
) => {
  return numerator / denominator;
};

export const calculationTypeMultiply = (value1: number, value2: number) => {
  return value1 * value2;
};

export const calculationTypeMinus = (value1: number, value2: number) => {
  return value1 - value2;
};

export const mathRoundToTwoDecimalsIfNeeded = (value: number) => {
  return Math.round((value + Number.EPSILON) * 100) / 100;
};
