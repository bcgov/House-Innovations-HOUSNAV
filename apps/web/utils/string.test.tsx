// 3rd part
import { describe, it, expect } from "vitest";
import { JSX } from "react";
// repo
import { getQuestion } from "@repo/data/useWalkthroughTestData";
// local
import {
  getStringFromComponents,
  parseStringToComponents,
  getAnswerValueDisplay,
} from "./string";
import { isArray } from "./typeChecking";
import { renderWithWalkthroughProvider } from "../tests/utils";

const AnswerValueTestComponent = ({ questionId }: { questionId?: string }) => {
  return <>{getAnswerValueDisplay(questionId)}</>;
};

describe("string", () => {
  /*
   * parseStringToComponents
   */
  it("parseStringToComponents: defined-term", () => {
    // get test data
    const questionData = getQuestion("P04");

    // check if we have questionText and throw error if we don't
    if (!questionData?.questionData.questionText) {
      assert.fail(
        "question data is missing question text - need to fix test data",
      );
    }

    // parse question string
    let components = parseStringToComponents(
      questionData?.questionData.questionText,
    );

    if (typeof components === "string") {
      assert.fail(
        "question text is missing defined term - need to fix test data",
      );
    }

    // check if components is an array or not
    if (!Array.isArray(components)) {
      components = [components];
    }

    const hasDefinedTerm = components.some((component) => {
      // ignore if component is a string
      if (typeof component === "string") {
        return false;
      }

      // check if component is DefinedTerm
      return component.type.name === "DefinedTerm";
    });

    // expect hasDefinedTerm to be true
    expect(hasDefinedTerm).toBeTruthy();
  });
  it("parseStringToComponents: no defined-term", () => {
    // parse question string
    const components = parseStringToComponents(
      "Question text without a defined term in it.",
    );

    let componentsArray: JSX.Element[] | string[] | (string | JSX.Element)[] =
      [];
    // check if components is an array or not
    if (!Array.isArray(components)) {
      componentsArray = [components];
    }

    const hasDefinedTerm = componentsArray.some((component) => {
      // ignore if component is a string
      if (typeof component === "string") {
        return false;
      }

      // check if component is DefinedTerm
      return component.type.name === "DefinedTerm";
    });

    // expect hasDefinedTerm to be false
    expect(hasDefinedTerm).toBeFalsy();
  });
  it("parseStringToComponents: pdf-download-link", () => {
    let components = parseStringToComponents(
      "<pdf-download-link>Download PDF</pdf-download-link>",
    );

    if (typeof components === "string") {
      assert.fail(
        "question text is missing pdf download link - need to fix test data",
      );
    }

    if (!isArray(components)) {
      components = [components];
    }

    const hasPDFDownloadLink = components.some((component) => {
      if (typeof component === "string") {
        return false;
      }
      return component.type.name === "PDFDownloadLink";
    });

    expect(hasPDFDownloadLink).toBeTruthy();
  });
  it("parseStringToComponents: no pdf-download-link", () => {
    const components = parseStringToComponents(
      "Question text without a pdf download link in it.",
    );

    let componentsArray: JSX.Element[] | string[] | (string | JSX.Element)[] =
      [];
    if (!isArray(components)) {
      componentsArray = [components];
    }

    const hasPDFDownloadLink = componentsArray.some((component) => {
      if (typeof component === "string") {
        return false;
      }
      return component.type.name === "PDFDownloadLink";
    });

    expect(hasPDFDownloadLink).toBeFalsy();
  });
  /*
   * getStringFromComponents
   */
  it("getStringFromComponents: string", () => {
    const string = "test string";
    const result = getStringFromComponents(string);
    expect(result).toBe(string);
  });
  it("getStringFromComponents: string with component", () => {
    const string = "test string";
    const stringWithComponent = "test <potato>string</potato>";
    const components = parseStringToComponents(stringWithComponent);
    const result = getStringFromComponents(components);
    expect(result).toBe(string);
  });
  it("getStringFromComponents: null", () => {
    const result = getStringFromComponents(null);
    expect(result).toBe("");
  });
  it("getStringFromComponents: number", () => {
    const number = 123;
    const result = getStringFromComponents(number);
    expect(result).toBe(number.toString());
  });
  it("getStringFromComponents: boolean", () => {
    const boolean = true;
    const result = getStringFromComponents(boolean);
    expect(result).toBe("");
  });
  /*
   * getAnswerValueDisplay
   */
  it("getAnswerValueDisplay: no questionId", () => {
    const result = getAnswerValueDisplay();
    expect(result).toMatchSnapshot();
  });
  it("getAnswerValueDisplay: no displayValue", () => {
    const { container } = renderWithWalkthroughProvider({
      ui: <AnswerValueTestComponent questionId={"test"} />,
    });
    expect(container).toMatchSnapshot();
  });
  it("getAnswerValueDisplay: with displayValue", () => {
    const { container } = renderWithWalkthroughProvider({
      ui: <AnswerValueTestComponent questionId={"P01"} />,
      initialAnswers: {
        P01: "yes",
      },
    });
    expect(container).toMatchSnapshot();
  });
});
