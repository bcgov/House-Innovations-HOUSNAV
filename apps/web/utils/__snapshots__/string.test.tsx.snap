// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`string > getAnswerValueDisplay: no displayValue 1`] = `
<div>
  <span
    aria-label="blank"
  >
    _____
  </span>
</div>
`;

exports[`string > getAnswerValueDisplay: no questionId 1`] = `<AnswerDisplayValuePlaceholder />`;

exports[`string > getAnswerValueDisplay: with displayValue 1`] = `
<div>
  Yes
</div>
`;
